### PR TITLE
nemo-extensions: new, 4.8.0+git20210203

### PR DESCRIPTION
**Pull request by Cyanoxygen**
Topic Description
-----------------

Add a new package `nemo-extensions` .

Various extensions for `nemo` , a Cinnamon file manager.

Package(s) Affected
-------------------

- `nemo-extensions` : new
- `stopit` : run-time dependency (PyPI package)
- `pymediainfo` : run-time dependency (PyPI) package

Security Update?
----------------

No

Build Order
-----------

- `stopit`
- `pymediainfo`
- `nemo-extensions` 

*Actually you can directly build `nemo-extensions` as it will build these two packages first.*

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
- [x] Architecture-independent `noarch`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
- [ ] Architecture-independent `noarch`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

Testing notice
--------

The following statement determines a good test result:

- All built extensions are enabled in Plugin Settings (Edit -> Preferences -> Plugins).
  image

- All built extensions will not crash the entire application. (Check by `NEMO_DEBUG=all nemo`)
  - Any error can be ignored if it does not lead to a crash, as it depends on a package that does not exist in non-amd64 archs.
- Context menu has more options than normal, e.g. `Compress...` , `Sign`, `Encrypt` .

**Need a good test result from non-amd64 architectures.**

Additional notice
--------

- Why do you drop `nemo-terminal` extension?

`nemo-terminal` is a Python module, which is based on vte, you can use it as a standalone application (by directly running it), but when you use it with Nemo, it will render Nemo unusable or cause it to crash, because of GLib's more advanced memory management or something.

You can feel it by seeing this:

image

image

image

### CODE REVIEW

**Comment by Cyanoxygen** [comment]
Testing
-------

OS: AOSC OS Core 8.0.2
Arch: amd64
WM: Cinnamon
Nemo: 4.8.4

Here's a screenshot showing these extensions running normally, with nemo 4.8.4.

image

Conclusion
----------

- [x] Test passed in `amd64`
- [x] Test passed in non-amd64
- [x] Ability to merge

File: extra-python/pymediainfo/spec [hunk 1]
```diff
@@ -0,0 +1,3 @@
+VER=5.0.3
+SRCS="https://files.pythonhosted.org/packages/1f/ec/da04fb8cbf5df9684b549bc3cb5c2010a0661da4bc626a6d531462222511/pymediainfo-${VER}.tar.gz"
```
Comment (MingcongBai) [inline extra-python/pymediainfo/spec]: You should use `pypi.io` instead. See `extra-python/april` to see how these links work.

File: extra-cinnamon/nemo-extensions/spec [hunk 2]
```diff
@@ -0,0 +1,4 @@
+VER=4.8.0+git20210203
+SRCS="git::commit=602111d7f038331354365826dbfa8af51ce141b7::https://github.com/linuxmint/nemo-extensions"
+CHKSUMS="SKIP"
+
```
Comment (MingcongBai) [inline extra-cinnamon/nemo-extensions/spec]: Trailing blank line.

File: extra-python/pymediainfo/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=pymediainfo
+PKGSEC=python
+PKGDES="A Python wrapper around the MediaInfo library"
+PKGDEP="python-3 libmediainfo"
+
+ABHOST=noarch
+ABTYPE=python
+NOPYTHON2=1
```
Comment (MingcongBai) [inline extra-python/pymediainfo/autobuild/defines]: Missing line ending.

File: extra-python/stopit/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=stopit
+PKGSEC=python
+PKGDES="Module to control the timeout of blocks or callables with a context manager or a decorator"
+PKGDEP="python-3"
+
+ABHOST=noarch
+ABTYPE=python
+NOPYTHON2=1
```
Comment (MingcongBai) [inline extra-python/stopit/autobuild/defines]: Missing line ending.

File: extra-cinnamon/nemo-extensions/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,15 @@
+PKGNAME=nemo-extensions
+PKGSEC=Cinnamon
+PKGDEP="python-3 nemo cinnamon clutter xreader gtksourceview-4 libmusicbrainz-5 freetype \
+	gdk-pixbuf cjs glib gstreamer-1-0 gtk-3 gobject-introspection webkit2gtk \
+	seahorse cinnamon-desktop samba file-roller polkit-gnome imagemagick \
+	pypdf2 pillow pastebinit mutagen pymediainfo stopit"
+
```
Comment (MingcongBai) [inline extra-cinnamon/nemo-extensions/autobuild/defines]: Probably not a problem but why a blank line here?

File: extra-cinnamon/nemo-extensions/autobuild/build [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+# Extensions using Meson build system
+MESONEXTS=()
+# Extensions using Python binding
+PYEXTS=()
+# Problematic extensions
+FAILEXTS=("nemo-terminal")
+
+pushd "$SRCDIR"
+
+for i in `find -maxdepth 1 -mindepth 1 -type d | sed 's/^.\///g'`; do
```
Comment (MingcongBai) [inline extra-cinnamon/nemo-extensions/autobuild/build]: Use absolute path here by appending `"$SRCDIR"` immediately after `find`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by Cyanoxygen** [comment]
Waiting for a good test result from non-amd64 arch.
(Will do this afternoon)

**Comment by Cyanoxygen** [comment]
@Cyanoxygen

**Comment by Cyanoxygen** [comment]
Testing (Non-amd64)
-------

OS: AOSC OS Core 8.0.2
Arch: arm64
WM: Cinnamon 
Nemo: 4.8.4

Here's the screenshot showing these extensions working well without an issue, under arm64.

image

Conclusion at https://github.com/AOSC-Dev/aosc-os-abbs/pull/2819#issuecomment-786385676

**Pull request merged**